webapi/src/main/scala/org/knora/webapi/util/InputValidation.scala

```diff
@@ -600,6 +614,91 @@ object InputValidation {
+                OntologyConstants.KnoraApi.ApiOntologyStart + apiOntologyName + OntologyConstants.KnoraApiV2WithValueObject.VersionSegment
+            case _ => errorFun()
+        }
+    }
+
+    /**
+      * Given the Iri of an external knora-api v2 with value object entity, returns the internal entity Iri.
+      *
+      * @param externalEntityIri an external entity Iri.
+      * @param errorFun a function that throws an exception. It will be called if the form of the string is not
+      *                 valid for an internal ontology IRI.
+      * @return the internal entity Iri.
+      */
+    def externalApiV2WithValueObjectEntityIriToInternalEntityIri(externalEntityIri: IRI, errorFun: () => Nothing): IRI = {
+        externalEntityIri match {
+            case ExternalApiV2WithValueObjectOntologyEntityRegex(ontology, entityName) =>
+                val ontologyName = if (ontology == "knora-api") "knora-base" else ontology
+                OntologyConstants.KnoraInternal.InternalOntologyStart + ontologyName + "#" + entityName
+            case _ => errorFun()
+        }
+    }
+
+    /**
+      * Given the IRI of an internal ontology entity, returns the knora-api v2 with value object entity Iri.
+      *
+      * @param internalEntityIri the Iri of the internal ontology entity.
+      * @param errorFun a function that throws an exception. It will be called if the form of the string is not
+      *                 valid for an internal ontology entity IRI.
+      * @return the corresponding knora-api v2 with value object entity Iri.
+      */
+    def internalEntityIriToApiV2WithValueObjectEntityIri(internalEntityIri: IRI, errorFun: () => Nothing): IRI = {
+        internalEntityIri match {
+            case ProjectSpecificInternalOntologyEntityRegex(prefixLabel, entityName) =>
+                val apiPrefixLabel = if (prefixLabel == "knora-base") "knora-api" else prefixLabel
+                OntologyConstants.KnoraApi.ApiOntologyStart + apiPrefixLabel + OntologyConstants.KnoraApiV2WithValueObject.VersionSegment + "#" + entityName
+            case _ => errorFun()
+        }
+    }
+
+    /**
+      * Given the IRI of an internal ontology entitiy, returns the simplified knora-api v2 entity Iri.
+      *
+      * @param internalEntityIri the Iri of the internal ontology entity.
+      * @param errorFun a function that throws an exception. It will be called if the form of the string is not
+      *                 valid for an internal ontology entity IRI.
+      * @return the corresponding simplified knora-api v2.
+      */
+    def internalEntityIriToSimpleApiV2EntityIri(internalEntityIri: IRI, errorFun: () => Nothing): IRI = {
+        internalEntityIri match {
+            case ProjectSpecificInternalOntologyEntityRegex(prefixLabel, entityName) =>
+                val apiPrefixLabel = if (prefixLabel == "knora-base") "knora-api" else prefixLabel
```

Comment:
Since the prefix label can be `knora-base`, the regex shouldn't be called `ProjectSpecificInternalOntologyEntityRegex`. Can we just rename it to `InternalOntologyEntityRegex`?